spec/requests/parents_request_spec.rb

```diff
@@ -0,0 +1,270 @@
+require 'rails_helper'
+
+RSpec.describe "Admin::Parents", type: :request do
+  let(:parent) { create(:parent) }
+  let(:parents) { create(:parent, 3) }
+  let!(:admin) { create(:user, :admin) }
+  let!(:user) { create(:user) }
+  let!(:not_user) { nil }
+  let!(:invalid_id) { '998' }
+  let(:not_parent) { nil }
+  let(:new_user) {create(:user)}
+
+  describe 'GET #index' do
+    context 'with logged-in admin' do
+      before do
+        sign_in admin
+        get admin_parents_path
+      end
+      it 'index status' do
+        expect(response).to have_http_status(:success)
+      end
+      it 'render index view' do
+        is_expected.to render_template :index
+      end
+    end
+
+    context 'with logged-in user' do
+      before do
+        sign_in user
+        get admin_parents_path
+      end
+      it 'index status' do
+        expect(response).to redirect_to root_path
+      end
+    end
+
+    context "not logged-in user" do
+      before do
+        sign_in :not_user
+        get admin_parents_path
+      end
+      it { expect(response.status).to eq(302) }
+      it { expect(response).to redirect_to new_user_session_path }
+    end
+  end
+
+  describe 'GET #show' do
+    context 'with logged-in admin' do
+      before do
+        sign_in admin
+        get admin_parent_path(id: parent.id)
+      end
+      it 'show status' do
+        expect(response).to have_http_status(:success)
+      end
+      it 'should show parent' do
+        is_expected.to render_template :show
+      end
+
+      context 'does not exist parent' do
```

Comment:
this context uses his own before hook and common before hook as a result, there are two sign-in actions and two requests to controller

Reply:
i tried to use before blocks where i could.